sn_node/tests/double_spend.rs

```diff
@@ -267,3 +270,134 @@ async fn poisoning_old_spend_should_not_affect_descendant() -> Result<()> {
+        wallet_c.address(),
+        DerivationIndex::random(&mut rng),
+    );
+    let transfer_to_c = OfflineTransfer::new(
+        cash_notes_b.clone(),
+        vec![to_c_unique_key],
+        wallet_b.address(),
+        reason.clone(),
+    )?;
+
+    client
+        .send_spends(transfer_to_c.all_spend_requests.iter(), false)
+        .await?;
+
+    info!("Verifying the transfers from B -> C wallet...");
+    let cash_notes_for_c: Vec<_> = transfer_to_c.cash_notes_for_recipient.clone();
+    client.verify_cashnote(&cash_notes_for_c[0]).await?;
+    wallet_c.deposit_and_store_to_disk(&cash_notes_for_c)?; // store inside c
+
+    // Try to double spend from A -> X
+    let wallet_dir_x = TempDir::new()?;
+    let wallet_x = get_wallet(wallet_dir_x.path());
+    assert_eq!(wallet_x.balance(), NanoTokens::zero());
+
+    let to_x_unique_key = (
+        amount,
+        wallet_x.address(),
+        DerivationIndex::random(&mut rng),
+    );
+    let transfer_to_x = OfflineTransfer::new(
+        cash_notes_a,
+        vec![to_x_unique_key],
+        wallet_a.address(),
+        reason.clone(),
+    )?; // reuse the old cash notes
+    client
+        .send_spends(transfer_to_x.all_spend_requests.iter(), false)
+        .await?;
+    info!("Verifying the transfers from A -> X wallet... It should error out.");
+    let cash_notes_for_x: Vec<_> = transfer_to_x.cash_notes_for_recipient.clone();
+    assert!(client.verify_cashnote(&cash_notes_for_x[0]).await.is_err()); // the old spend has been poisoned
+
+    // Try to double spend from B -> Y
+    let wallet_dir_y = TempDir::new()?;
+    let wallet_y = get_wallet(wallet_dir_y.path());
+    assert_eq!(wallet_y.balance(), NanoTokens::zero());
+
+    let to_y_unique_key = (
+        amount,
+        wallet_y.address(),
+        DerivationIndex::random(&mut rng),
+    );
+    let transfer_to_y = OfflineTransfer::new(
+        cash_notes_b,
+        vec![to_y_unique_key],
+        wallet_b.address(),
+        reason.clone(),
+    )?; // reuse the old cash notes
+    client
+        .send_spends(transfer_to_y.all_spend_requests.iter(), false)
+        .await?;
+    info!("Verifying the transfers from B -> Y wallet... It should error out.");
+    let cash_notes_for_y: Vec<_> = transfer_to_y.cash_notes_for_recipient.clone();
+    let result = client.verify_cashnote(&cash_notes_for_y[0]).await;
+    info!("Got result while verifying double spend from B -> Y: {result:?}");
+    assert!(result.is_err());
+
+    info!("Verifying the original cashnote of A -> B");
+    let result = client.verify_cashnote(&cash_notes_for_b[0]).await;
+    info!("Got result while verifying the original spend from A -> B: {result:?}");
+    assert!(result.is_err());
+
+    info!("Verifying the original cashnote of B -> C");
+    let result = client.verify_cashnote(&cash_notes_for_c[0]).await;
+    info!("Got result while verifying the original spend from B -> C: {result:?}");
+    assert!(result.is_err());
```

Comment:
Maybe it's better if we specify the exact error type we're expecting so we're sure why we error out?

Reply:
Right yeah! Will do that.